packages/core/src/browser/keybinding.ts

```diff
@@ -132,60 +131,166 @@ export class KeybindingRegistry {
         for (const context of contexts) {
             const { id } = context;
             if (this.contexts[id]) {
-                console.error(`A keybinding context with ID ${id} is already registered.`);
+                this.logger.error(`A keybinding context with ID ${id} is already registered.`);
             } else {
                 this.contexts[id] = context;
             }
         }
     }
 
+    /**
+     * Register a default keybinding to the registry.
+     *
+     * @param binding
+     */
+    registerKeybinding(binding: Keybinding) {
+        this.doRegisterKeybinding(binding, KeybindingScope.DEFAULT);
+    }
+
+    /**
+     * Register default keybindings to the registry
+     *
+     * @param bindings
+     */
     registerKeybindings(...bindings: Keybinding[]): void {
+        this.doRegisterKeybindings(bindings, KeybindingScope.DEFAULT);
+    }
+
+    protected doRegisterKeybindings(bindings: Keybinding[], scope: KeybindingScope = KeybindingScope.DEFAULT) {
         for (const binding of bindings) {
-            this.registerKeybinding(binding);
+            this.doRegisterKeybinding(binding, scope);
+        }
+    }
+
+    protected doRegisterKeybinding(binding: Keybinding, scope: KeybindingScope = KeybindingScope.DEFAULT) {
+        try {
+            if (!this.isKeybindingAvailable(this.keymaps[scope], binding)) {
+                throw new Error(`"${binding.keybinding}" is in collision with something else [scope:${scope}]`);
+            }
+            this.keymaps[scope].push(binding);
+        } catch (error) {
+            this.logger.warn(`Could not register keybinding:\n  ${Keybinding.stringify(binding)}\n${error}`);
         }
     }
 
     /**
-     * Register a default keybinding to the registry.
+     * Checks for keySequence collisions in a list of Keybindings
      *
-     * @param binding
+     * @param bindings the keybinding reference list
+     * @param binding the keybinding to test collisions for
      */
-    registerKeybinding(binding: Keybinding) {
+    isKeybindingAvailable(bindings: Keybinding[], binding: Keybinding): boolean {
```

Comment:
This probably needs a better name `isKeyBindingAvailable` does not really explain what this function does.

Reply:
Well, it checks whether or not a keybinding is available within a given keybinding collection... I would stick to this, do you have an idea for the name ?

Reply:
@gorkem do you still want this to be changed ?

Reply:
`containsKeybinding` might be more appropriate.